File: code-examples/scala/usage-examples/updateMany.scala
import org.mongodb.scala._

object UpdateQueries {
  val mongoClient: MongoClient = MongoClient("<connection string uri>")
  val database: MongoDatabase = mongoClient.getDatabase("sample_mflix")
  val collection: MongoCollection[Document] = database.getCollection("movies")

  sealed trait Rating
  case object G extends Rating
  case object PG extends Rating
  case object PG_13 extends Rating
  case object R extends Rating
  case object NR extends Rating

  case class Movie(rated: Rating, random_review: Option[String] = None)

  def giveRandomReview(): String = 
    s"After viewing I am ${100 * scala.util.Random.nextDouble()}% more satisfied with life."

  def updateMovies() {
    val filter = Document("rated" -> G.toString())
    val update = Document("$set" -> Document("random_review" -> giveRandomReview()))
    
    collection.updateMany(filter, update).subscribe(
      (result: UpdateResult) => println(s"Updated ${result.getModifiedCount} documents"),
      (t: Throwable) => t.printStackTrace()
    )
  }

  def closeConnection() {
      mongoClient.close()
  }
}

object Main extends App {
  UpdateQueries.updateMovies()
  UpdateQueries.closeConnection()
}

File: code-examples/java-rs/usage-examples/updateMany.java
import com.mongodb.MongoClient;
import com.mongodb.reactiverestreams.client.MongoDatabase;
import com.mongodb.reactiverestreams.client.MongoCollection;
import com.mongodb.client.model.Filters;
import com.mongodb.client.model.Updates;
import org.bson.Document;
import java.util.concurrent.TimeUnit;
import java.util.Random;

public class MovieUpdateSample {
  enum Rating {
    G, PG, PG_13, R, NR;
  }

  static class Movie {
    Rating rated;
    String randomReview;
    
    // getters and setters
  }

  public static void main(String[] args) {
    MongoClient mongoClient = new MongoClient("<connection string uri>");
    MongoDatabase database = mongoClient.getDatabase("sample_mflix");
    MongoCollection<Document> movies = database.getCollection("movies");

    try {
      movies.updateMany(
        Filters.eq("rated", Rating.G.name()),
        Updates.set("random_review", "After viewing I am " + new Random().nextFloat() * 100 + "% more satisfied with life.")
      ).subscribe(updateResult -> {
            System.out.println("Updated " + updateResult.getModifiedCount() + " documents");
        }, error -> {
            System.err.println("An error occurred.");
            error.printStackTrace();
        }, () -> {
            // Close connection in onComplete of operation.
            mongoClient.close();
        });
    } catch (Exception e) {
      e.printStackTrace();
    } finally {
      // Close connection on completion or error after a timeout.
      mongoClient.close(10, TimeUnit.SECONDS);
    }
  }
}
